### ISSUE TITLE
[Feature] Front matter boolean for disabling hero image per post page

### ISSUE THREAD

**Issue by bravo-kernel** [narative/gatsby-theme-novela#43]
It would be nice if we could optionally turn off the hero per page. E.g.

```yaml
hero: ./images/hero.jpg
renderPostHero: false
```
Should probably be `true` by default and when unspecified.

**Comment by brotzky** [comment]
Yes, will think about this one. We can't disable images on the home page as it's not designed that way yet, but on the article page it might be a good idea.

You're not the first to request this so we'll investigate it today.

**Comment by bravo-kernel** [comment]
Nice and that's exactly what I mean... display the hero on the home page (as that looks nice) but disable per article (if specified).

**Comment by bravo-kernel** [comment]
Implemented here https://github.com/ryanwiemer/gatsby-theme-amsterdam/pull/5

**Comment by brotzky** [comment]
Would love a PR for this one. The design is very geared to including the image so ill consider this out of scope of what we'll do for now. Use nice hero images you'd want to display :D

**Comment by bravo-kernel** [comment]
I'm having a bit of a hard time figuring out how to PR. How do you folks do this? I was hoping to clone the theme, then run it locally so I can see the changes while programming. Done.. just commit and push.

**Comment by bravo-kernel** [comment]
I will add some documentation once I figure this out.

**Comment by brotzky** [comment]
Hit Fork
now you have your own copy
clone that locally and run yarn install
`www` is where you'll edit your "site"
`@narative/gatsby-theme-novela` is where you'll configure the theme
commit and push what you've changed
go to your forked Github repo and there should be a message saying create a pull request
if not, come back to this repo and see if the message is there.

**Comment by bravo-kernel** [comment]
Wow, I think this has been too long a day. It does work exactly as I expected.

Here's what I am doing:
```
git clone git@github.com:bravo-kernel/gatsby-theme-novela.git
cd gatsby-theme-novela
yarn workspace www install
yarn dev
```

- This starts the www site as expected at `localhost:8000` 
- I now make a change in file `@narative/gatsby-theme-novela/`
- Change in `src` is live reload in the www site

Sorry for the noise. I will pick up on this tomorrow.

**Comment by brotzky** [comment]
No problem, thanks for all the help.
I'm having a long day today as well 😅

**Comment by brotzky** [comment]
@bravo-kernel I think https://github.com/narative/gatsby-theme-novela/pull/62 solves this as well.

an image will always be required with the design of this version of Novela and I think allowing authoring without an image is a great compromise. Let me know if you had something else in mind.

**Comment by bravo-kernel** [comment]
@brotzky I was just about to test this when I noticed `yarn dev` doesn't start anymore, looks to be caused by Contentful. I tried changing contentful to `false` like shown below but same result.

Please do point me in right direction if I am misinterpreting stuff again, I thought Contentful was optional.

```js
        sources: {
          local: true,
          contentful: false
        },
```

**Comment by brotzky** [comment]
What was the error?

It should work without explicitly setting those values.

**Comment by bravo-kernel** [comment]
I made a local change that adds this functionality. There were however two things that I could not figure out yet:

- the Aside progressbar calculations 
- the css (it now just displays an empty block)

I will probably have to dig in to React a little deeper this weekend but it looks very doable to me. IMHO you will have a clear winner once this feature lands as it would make the theme suitable to most use-cases. If needed I can PR anyway, perhaps with some pointers I will be able to get it right.

**Comment by brotzky** [comment]
PR is appreciated, even to share the initial ideas.

**Comment by bravo-kernel** [comment]
@brotzky apologies for the late reply, I had to push out a new library which took more time than I expected. 

In all honesty, my PR would only show a frontmatter similar to 'secret' but would not solve anything related to A-Side and CSS (since I am no Gatsby/React pro yet). **How about this?** You implement this feature and I start work on #52, should be do-able today.

Also some more background information on this request. The migration script for my current blog aims to produce the following:

## without-image-render-on-post

Applied to all articles matching a certain "Category" with end-result:

- the same generic category-image used for all posts matching the category
- this category-image makes the posts easily identifiable as a group on the landing page
- when opening the post we do not show the category-image (no added value) but jump straight into the content (mostly tutorials)

## with-image-render-on-post

This would apply to all non-categorized/unique posts:

- unique image per post
- for example my 'Perfect Cosmopolitan' post
- this would show an image of the cocktail on both landing and the article page (as it is unique for that post and adds value)

Hope this makes sense.

**Comment by brotzky** [comment]
Im going to stand that we keep images on post pages for now. I dont want to ruin the integrity of the design and there's already an image required for the aritcles list.

**Comment by bravo-kernel** [comment]
Ahh, disappointed to hear that but you probably have good reasons. Out of curiosity.. was it the complexity of the change?

**Comment by brotzky** [comment]
Want to avoid going too far from core design.

**Comment by bravo-kernel** [comment]
ok, fair enough
